app/containers/ActionSheet/ActionSheet.tsx

```diff
@@ -39,27 +28,26 @@ const ANIMATION_CONFIG = {
 
 const ActionSheet = React.memo(
 	forwardRef(({ children }: { children: React.ReactElement }, ref) => {
-		const { theme } = useTheme();
-		const bottomSheetRef = useRef<ScrollBottomSheet<TActionSheetOptionsItem>>(null);
+		const { colors } = useTheme();
+		const bottomSheetRef = useRef<BottomSheet>(null);
 		const [data, setData] = useState<TActionSheetOptions>({} as TActionSheetOptions);
 		const [isVisible, setVisible] = useState(false);
 		const { height } = useDimensions();
 		const { isLandscape } = useOrientation();
 		const insets = useSafeAreaInsets();
 
-		const maxSnap = Math.max(
-			height -
-				// Items height
-				ITEM_HEIGHT * (data?.options?.length || 0) -
+		const maxSnap = Math.min(
```

Comment:
We could simplify this logic by using [useBottomSheetDynamicSnapPoints])(https://gorhom.github.io/react-native-bottom-sheet/hooks#usebottomsheetdynamicsnappoints).
This code here is working (your proposal too), but it's using lots of static info.
Maybe we could use that hook, remove this complex logic from our component and only say our snaps are `['50%', 'CONTENT_HEIGHT']`